mbed_tools_ci_scripts/license_files.py

```diff
@@ -0,0 +1,85 @@
+#
+# Copyright (c) 2020-2020, Arm Mbed. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+#
+"""Apply copyright and licensing to all source files present in a project.
+
+This is to comply with OpenChain certification;
+https://github.com/OpenChain-Project/Curriculum/blob/master/guides/reusing_software.md#2-include-a-copyright-notice-and-license-in-each-file
+"""
+import argparse
+import logging
+import subprocess
+import sys
+import tempfile
+from datetime import datetime
+from mbed_tools_ci_scripts.utils.configuration import configuration, ConfigurationVariable
+from mbed_tools_ci_scripts.utils.logging import set_log_level, log_exception
+from pathlib import Path
+
+logger = logging.getLogger(__name__)
+
+LICENCE_HEADER_TEMPLATE = """Copyright (c){date}, {author}. All rights reserved.
+SPDX-License-Identifier: {licence_identifier}
+"""
+
+
+class CopyrightNotifierToolWrapper:
+    """Wrapper over the licenseheaders tool.
+
+    See https://github.com/johann-petrak/licenseheader for more details about the tool in use.
+    copyright (https://github.com/knipknap/copyright) was first considered but
+    comprises quite a few bugs and does not seem active anymore.
```

Comment:
Is this the right place to put this? Feels more like an architectural decision to me, we need a record, but users don't necessarily need to know.

Reply:
moved this out of the docstring